stdlib/src/memory/unsafe.mojo

```diff
@@ -221,6 +221,9 @@ struct LegacyPointer[
         )
         return Self {address: address}
 
+    fn __init__(p: UnsafePointer[type, address_space]) -> Self:
+        return Self(p.value)
+
```

Comment:
Unfortunately it is not safe in general to construct a `LegacyPointer` from an `UnsafePointer` because `LegacyPointer` is parameterized on a `type: AnyRegType`, which will effectively type-erase important information about the pointee type.

We also don't generally want to make it easy to go back from an `UnsafePointer` to a `LegacyPointer`.

I may have missed something, but I think this new initializer is not being used in this PR? If so, I suggest we remove it.

Otherwise I'm happy to get a better understanding of what the intended use-cases are 🙂

Reply:
I added this in case the changed return type of `_malloc` becomes a problem. Perhaps there is nothing to worry about.

Reply:
I think so, lets hold off. 

If the changed return type of `_malloc` does cause any problems I think those would be better handled on a case-by-case basis (and it would be good to get a compiler error so we know where that happens).

Thanks @soraros!